src/LocationPickerWidget.js

```diff
@@ -128,7 +211,7 @@ export default class LocationPickerWidget extends React.Component {
           };
 
           const valueJSON = JSON.stringify(location);
-
+          geocoderControl.setInput(location.address);
```

Comment:
honestly this seems like a pretty clean solution despite it feeling hacky, as long as we do a lot of testing it might work out nicely